content/en/docs/tasks/debug/debug-cluster/troubleshoot-kubectl.md

```diff
@@ -0,0 +1,144 @@
+to identify and address the likely cause.
+
+<!-- body -->
+
+## {{% heading "prerequisites" %}}
+
+* You need to have a Kubernetes cluster.
+* You also need to have `kubcectl` installed.
+
+## Verify kubectl setup
+
+Make sure you have installed and configured `kubectl` correctly on your local machine.
+Check the `kubectl` version to ensure it is up-to-date and compatible with your cluster.
+
+Check kubectl version:
+
+```shell
+kubectl version
+```
+
+You'll see a similar output:
+
+```console
+Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4",GitCommit:"fa3d7990104d7c1f16943a67f11b154b71f6a132", GitTreeState:"clean",BuildDate:"2023-07-19T12:20:54Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
+Kustomize Version: v5.0.1
+Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3",GitCommit:"25b4e43193bcda6c7328a6d147b1fb73a33f1598", GitTreeState:"clean",BuildDate:"2023-06-14T09:47:40Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}
+
+```
+
+If you see `Unable to connect to the server: dial tcp <server-ip>:8443: i/o timeout`,
+instead of `Server Version`, you need to troubleshoot kubectl connectivity with your cluster.
+
+If you have installed the `kubectl` but your shell gives the `command not found` error
+make sure that the `$PATH` environment variable is correctly configured and you have
+installed the kubectl by following the [official documentation for installing kubectl](/docs/tasks/tools/#kubectl).
+
+## Check VPN connectivity
+
+If you are using a Virtual Private Network (VPN) to access your Kubernetes cluster,
+make sure that your VPN connection is active and stable. Sometimes, VPN disconnections
+can lead to connection issues with the cluster. Reconnect to the VPN and try accessing
+the cluster again.
+
+## Check kubeconfig
+
+The `kubectl` requires a `kubeconfig` file to connect to a Kubernetes cluster. The
+`kubeconfig` file is usually located under the `~/.kube/config` directory. Make sure
+that you have a valid `kubeconfig` file. If you don't have a `kubeconfig` file, you can
+create and configure it via `kubectl config` command. If you have deployed your Kubernetes
+cluster on a cloud platform and lost your `kubeconfig` file, you can re-generate it using
+your cloud provider's tools. Refer the cloud provider's documentation for re-generating
+the `kubeconfig` file.
+
+Check if the `$KUBECONFIG` environment variable is configured correctly. You can set
+`$KUBECONFIG`environment variable or use the `--kubeconfig` parameter with the kubectl
+to specify the directory of a `kubeconfig` file.
+
+## Authentication and authorization
+
+If the kubectl is returning an error regarding the authentication token or authentication
+server, validate the `$KUBERNETES_SERVER` and`$KUBERNETES_TOKEN` environment variables.
+
+Make sure that you are using the valid user credentials. And you have the permission to
+access the resource that you have requested.
```

Comment:
The second sentence comes from no where. I haven't even try to access any resources.
What kind of an error will show up when I don't have permissions?
As a user, I need to tell the difference between authentication error and authorization error.